### PR TITLE
Migrate YDB Operator to new DBAPI

### PR DESCRIPTION
**Pull request by vgvoleg**
For historical reasons, the YDB operator had a copied DBAPI from the ydb-platform/ydb-sqlalchemy repository. It was impossible to use it without copying - the repository has a dependency on SQLAlchemy > 2.0. (problem was described here https://github.com/apache/airflow/blob/995cd8fba8de4ab3d041c7e141a0b92d786a8277/providers/src/airflow/providers/ydb/hooks/_vendor/readme.md)

At the moment we split the DBAPI and SQLAlchemy Dialect, so now there are no conflicts between them. 

In this pull request, we delete the copied files, adding a new dependency on the ydb-dbapi package, and also adapt the current YDB Operator to the new dbapi. 

YDBScanQueryOperator was more of an experiment, the new YDBExecuteQueryOperator on top of the new dbapi covers all scenarios - no need to stop it (as far as I know, none of our users have started using it)

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: providers/src/airflow/providers/ydb/hooks/ydb.py [hunk 1]
```diff
@@ -136,10 +138,9 @@ class YDBHook(DbApiHook):
     supports_autocommit: bool = True
     supports_executemany: bool = True
 
-    def __init__(self, *args, is_ddl: bool = False, use_scan_query: bool = False, **kwargs) -> None:
+    def __init__(self, *args, is_ddl: bool = False, **kwargs) -> None:
         super().__init__(*args, **kwargs)
         self.is_ddl = is_ddl
-        self.use_scan_query = use_scan_query
 
         conn: Connection = self.get_connection(self.get_conn_id())
```
Comment (dabla) [inline providers/src/airflow/providers/ydb/hooks/ydb.py]: I would use the self.connection property here, as this is a cached shortcut of  self.get_connection(self.get_conn_id()) which here would make the code more readable.  This is a shortcut property that has been added lately in the DbApiHook

Comment (dabla) [reply providers/src/airflow/providers/ydb/hooks/ydb.py]: I really like the clean up, nice work @vgvoleg

**Review by dabla — CHANGES_REQUESTED** [review]

File: providers/src/airflow/providers/ydb/hooks/ydb.py (hunk 1 above)
Comment (vgvoleg) [reply providers/src/airflow/providers/ydb/hooks/ydb.py]: @dabla thanks for you review! 
I've fixed this, nice one.

File: providers/src/airflow/providers/ydb/hooks/ydb.py [hunk 2]
```diff
@@ -136,12 +139,11 @@ class YDBHook(DbApiHook):
     supports_autocommit: bool = True
     supports_executemany: bool = True
 
-    def __init__(self, *args, is_ddl: bool = False, use_scan_query: bool = False, **kwargs) -> None:
+    def __init__(self, *args, is_ddl: bool = False, **kwargs) -> None:
```
Comment (potiuk) [inline providers/src/airflow/providers/ydb/hooks/ydb.py:142]: Those all look like breaking changes, can you make a note at the top of ydb Changelog explaining that this is a breaking changes and summarizing changes for the users ?

Then release manager will apply version change when preparing the documentation.

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
Nice set of changes BTW :)

**Review by potiuk — APPROVED** [review]

File: providers/src/airflow/providers/ydb/CHANGELOG.rst [hunk 3]
```diff
@@ -27,6 +27,10 @@
 Changelog
 ---------
 
+.. note::
+  This release removes YDBScanQueryOperator from this provider package.
+  At this point, YDBExecuteQueryOperator could load unlimited amount of rows, so no speific operator is needed.
```
Comment (uzhastik) [inline providers/src/airflow/providers/ydb/CHANGELOG.rst]: Typo in speific

Comment (potiuk) [reply providers/src/airflow/providers/ydb/CHANGELOG.rst]: Our spellchecker is much more picky than maintainers :).

**Review by uzhastik — APPROVED** [review]

File: providers/src/airflow/providers/ydb/CHANGELOG.rst [hunk 4]
```diff
@@ -27,6 +27,10 @@
 Changelog
 ---------
 
+.. note::
+  This release removes YDBScanQueryOperator from this provider package.
+  At this point, YDBExecuteQueryOperator could load unlimited amount of rows, so no specific operator is needed.
+
```
Comment (eladkal) [inline providers/src/airflow/providers/ydb/CHANGELOG.rst:33]: is this a breaking change?

Comment (vgvoleg) [reply providers/src/airflow/providers/ydb/CHANGELOG.rst:33]: Yes, we've deleted operator that was added in prev release. I'm pretty sure that no customers used it.

**Pull request merged**